### PR TITLE
Nvli 178

### PR DESCRIPTION
**Pull request by dishabhadra**
@prateekjain  : Review the code and then kindly merge my branch.

### CODE REVIEW

**Comment by dishabhadra** [comment]
@piyuesh23 : All the changes as per your review are done. Please review it again.

**Comment by dishabhadra** [comment]
I am harcoding the format as backend level as of now because we are not having config entity.

File: docroot/modules/custom/nvli_custom_search_api/src/Plugin/rest/resource/NvliSearchResource.php [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+   *
+   * @return \Drupal\rest\ResourceResponse
+   *   The HTTP response object.
+   *
+   * @throws \Symfony\Component\HttpKernel\Exception\HttpException
+   *   Throws exception expected.
+   */
+  public function get() {
+// Fetch the query parameters.
+    $offset = \Drupal::request()->get('offset', $default);
```
Comment (piyuesh23) [inline docroot/modules/custom/nvli_custom_search_api/src/Plugin/rest/resource/NvliSearchResource.php]: Where are we declaring this variable $default?

File: docroot/modules/custom/nvli_custom_search_api/src/Plugin/rest/resource/NvliSearchResource.php [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    $this->search = $search;
+    $this->searchall = $searchall;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+        $configuration, $plugin_id, $plugin_definition, $container->getParameter('serializer.formats'), $container->get('logger.factory')->get('rest'), $container->get('current_user'), $container->get('custom_solr_search.search'), $container->get('custom_solr_search.search_all')
```
Comment (piyuesh23) [inline docroot/modules/custom/nvli_custom_search_api/src/Plugin/rest/resource/NvliSearchResource.php]: Can you format this to have one item per line?

File: docroot/modules/custom/nvli_custom_search_api/src/Plugin/rest/resource/NvliSearchResource.php (hunk 1 above)
Comment (dishabhadra) [inline docroot/modules/custom/nvli_custom_search_api/src/Plugin/rest/resource/NvliSearchResource.php]: No need to declare $default. I removed it.

**Comment by dishabhadra** [comment]
@piyuesh23 : Last two comment changes are being done. Review it.

**Pull request merged**